app/src/main/java/com/github/amusinamaria/ui/details/DetailsFragment.kt

```diff
@@ -4,38 +4,46 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.ImageView
 import androidx.fragment.app.Fragment
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
-import com.github.amusinamaria.R
-import com.github.amusinamaria.repository.MockDataSource
+import com.github.amusinamaria.databinding.FragmentDetailsBinding
+import com.github.amusinamaria.repository.MovieCard
+import com.github.amusinamaria.ui.MainActivity.Companion.MOVIE_ARGS_KEY
 
 class DetailsFragment : Fragment() {
 
-    private lateinit var castRecycler: RecyclerView
+    private var _binding: FragmentDetailsBinding? = null
+    private val binding get() = _binding!!
     private lateinit var castAdapter: CastAdapter
+    private lateinit var movieCard: MovieCard
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? = inflater.inflate(R.layout.fragment_movies_details, container, false)
+    ): View {
+        _binding = FragmentDetailsBinding.inflate(inflater, container, false)
+        return binding.root
+    }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        view.findViewById<ImageView>(R.id.backArrow).setOnClickListener {
+        movieCard = arguments?.getParcelable(MOVIE_ARGS_KEY)!!
+        val movieTitle = binding.detailsTitle
+        movieTitle.text = movieCard.title
+
+        binding.backArrow.setOnClickListener {
             fragmentManager?.popBackStack()
         }
-        castRecycler = view.findViewById(R.id.castRecycler)
-        castRecycler.apply {
+        binding.castRecycler.apply {
             layoutManager = LinearLayoutManager(context, RecyclerView.HORIZONTAL, false)
             setHasFixedSize(true)
         }
         castAdapter = CastAdapter().apply {
             setHasStableIds(true)
         }
-        castRecycler.adapter = castAdapter
+        binding.castRecycler.adapter = castAdapter
```

Comment:
try to use binding.apply {}. for reducing the count of "binding."

Reply:
Done